student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java

```diff
@@ -149,17 +149,29 @@ public void approve(OpportunityInstance oppInstance, List<String> segmentsAccomm
         }
       }
     }
-  
+
     // Populate the accommodation dependencies for the *assessment* accommodation
     if (!segmentPositionToAccommodations.isEmpty()) {
       Accommodations assessmentAccommodation = segmentPositionToAccommodations.get(ASSESSMENT_POSITION);
-    
+
       for (AccommodationDependency assessmentDependency : assessment.getAccommodationDependencies()) {
         assessmentAccommodation.AddDependency(assessmentDependency.getIfType(), assessmentDependency.getIfValue(),
           assessmentDependency.getThenType(), assessmentDependency.getThenValue(), assessmentDependency.isDefault());
       }
     }
-    
+
+    // If this list does not contain the total number of segments + the parent assessment, we need to add a default Accommodations
+    if (segmentPositionToAccommodations.size() < assessment.getSegments().size() + 1) {
```

Comment:
I don't think those are logical equivalents. We want to check if the number of accommodations assigned to a segment position is less than the number of segments + 1 (the 1 represents the "assessment" accommodations").

As a side note, this wasn't meant to be included in this PR as it was a bug fix I was briefly investigating and thought I stashed - it accidentally snuck its way in.